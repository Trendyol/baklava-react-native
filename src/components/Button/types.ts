export type ButtonVariantVariantTypes = 'primary' | 'secondary' | 'tertiary';

export type ButtonKindVariantTypes =
  | 'default'
  | 'neutral'
  | 'success'
  | 'danger';

export type ButtonSizeVariantTypes = 's' | 'm' | 'l';
