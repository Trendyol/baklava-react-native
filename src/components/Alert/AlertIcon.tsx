import React from 'react';
import Icon from '../Icon/Icon';
import { IconNameType } from '../Icon/types';
import { VariantProps } from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

export const AlertIcon = React.memo(
  ({
    variant = 'info',
    icon,
  }: {
    icon?: IconNameType | boolean | null;
    variant?: VariantProps<Theme, 'alertVariants'>['variant'];
  }): JSX.Element | null => {
    const iconName = React.useMemo(() => {
      if (typeof icon === 'boolean' && !icon) {
        return null;
      }

      return {
        info: 'info',
        success: 'check-fill',
        warning: 'warning',
        danger: 'alert',
      }[variant];
    }, [variant, icon]) as IconNameType | null;

    const iconColor = React.useMemo(() => {
      return {
        info: 'primaryKey',
        warning: 'warningKey',
        success: 'successKey',
        danger: 'dangerKey',
      }[variant];
    }, [variant]);

    if (!iconName) {
      return null;
    }

    return (
      <Icon size="s" name={iconName} color={iconColor} testID="alert-icon" />
    );
  },
);
