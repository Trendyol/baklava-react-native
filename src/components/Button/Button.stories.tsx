import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Button from './Button';
import Box from '../Box/Box';
import Text from '../Text/Text';
import {
  ButtonSizeVariantTypes,
  ButtonKindVariantTypes,
  ButtonVariantVariantTypes,
} from './types';

const variantList = ['primary', 'secondary', 'tertiary'];
const kindList = ['default', 'neutral', 'success', 'danger'];
const sizeList = ['s', 'm', 'l'];

const ButtonMeta: ComponentMeta<typeof Button> = {
  title: 'Button',
  component: Button,
  argTypes: {
    onPress: { action: 'pressed the button' },
    variant: {
      options: variantList,
      control: { type: 'radios' },
    },
    kind: {
      options: kindList,
      control: { type: 'radios' },
    },
    size: {
      options: sizeList,
      control: { type: 'radios' },
    },
  },
  args: {
    size: sizeList[1] as ButtonSizeVariantTypes,
    kind: kindList[0] as ButtonKindVariantTypes,
    variant: variantList[0] as ButtonVariantVariantTypes,
    disabled: false,
    filled: false,
  },
};

export default ButtonMeta;

type ButtonStory = ComponentStory<typeof Button>;

export const Basic: ButtonStory = args => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Button
    </Text>
    <Button
      {...args}
      size={args.size}
      kind={args.kind}
      variant={args.variant}
      filled={args.filled}
      icon="confetti"
      label={
        args.disabled
          ? `disabled ${args.kind} ${args.size} button`
          : `${args.variant} ${args.kind} ${args.size} button`
      }
      m="2xs"
    />
  </>
);

export const ButtonVariants: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Button Variants
    </Text>
    <Box>
      <Button variant="primary" label="Primary Button" m="2xs" />
      <Button variant="secondary" label="Secondary Button" m="2xs" />
      <Button variant="tertiary" label="Tertiary Button" m="2xs" />
    </Box>
  </>
);

export const PrimaryButtons: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Primary Buttons
    </Text>
    <Box>
      <Button label="Primary Button" m="2xs" />
      <Button kind="neutral" label="Primary Button" m="2xs" />
      <Button kind="success" label="Primary Button" m="2xs" />
      <Button kind="danger" label="Primary Button" m="2xs" />
    </Box>
  </>
);

export const SecondaryButtons: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Secondary Buttons
    </Text>
    <Box>
      <Button variant="secondary" label="Secondary Button" m="2xs" />
      <Button
        variant="secondary"
        kind="neutral"
        label="Secondary Button"
        m="2xs"
      />
      <Button
        variant="secondary"
        kind="success"
        label="Secondary Button"
        m="2xs"
      />
      <Button
        variant="secondary"
        kind="danger"
        label="Secondary Button"
        m="2xs"
      />
    </Box>
  </>
);

export const TertiaryButtons: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Tertiary Buttons
    </Text>
    <Box>
      <Button variant="tertiary" label="Tertiary Button" m="2xs" />
      <Button
        variant="tertiary"
        kind="neutral"
        label="Tertiary Button"
        m="2xs"
      />
      <Button
        variant="tertiary"
        kind="success"
        label="Tertiary Button"
        m="2xs"
      />
      <Button
        variant="tertiary"
        kind="danger"
        label="Tertiary Button"
        m="2xs"
      />
    </Box>
  </>
);

export const IconButtons: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Icon Buttons
    </Text>
    <Box flexDirection="row" justifyContent="space-around">
      <Box>
        <Button label="Save" icon="info" m="2xs" />
        <Button kind="neutral" label="Save" icon="info" m="2xs" />
        <Button kind="success" label="Save" icon="info" m="2xs" />
        <Button kind="danger" label="Save" icon="info" m="2xs" />
      </Box>
      <Box>
        <Button variant="tertiary" label="Save" icon="info" m="2xs" />
        <Button
          variant="tertiary"
          kind="neutral"
          label="Save"
          icon="info"
          m="2xs"
        />
        <Button
          variant="tertiary"
          kind="success"
          label="Save"
          icon="info"
          m="2xs"
        />
        <Button
          variant="tertiary"
          kind="danger"
          label="Save"
          icon="info"
          m="2xs"
        />
      </Box>
    </Box>
  </>
);

export const IconOnlyButtons: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Icon Only Buttons
    </Text>
    <Box>
      <Button icon="info" m="2xs" />
      <Button kind="neutral" icon="info" m="2xs" />
      <Button kind="success" icon="info" m="2xs" />
      <Button kind="danger" icon="info" m="2xs" />
    </Box>
  </>
);

export const Sizes: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Button Sizes
    </Text>
    <Box>
      <Button size="l" icon="info" label="Primary Button" m="2xs" />
      <Button size="m" icon="info" label="Primary Button" m="2xs" />
      <Button size="s" icon="info" label="Primary Button" m="2xs" mb="2xl" />
      <Button
        size="l"
        icon="info"
        label="Primary Button"
        filled={true}
        m="2xs"
      />
      <Button
        size="m"
        icon="info"
        label="Primary Button"
        filled={true}
        m="2xs"
      />
      <Button
        size="s"
        icon="info"
        label="Primary Button"
        filled={true}
        m="2xs"
      />
    </Box>
  </>
);

export const DisabledButtons: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Disabled Buttons
    </Text>
    <Box mt="xl">
      <Text p="2xs" variant="captionMedium">
        Primary & Secondary Variants
      </Text>
      <Button size="l" disabled={true} label="Passive Button" m="2xs" />
      <Button size="m" disabled={true} label="Passive Button" m="2xs" />
      <Button
        size="s"
        disabled={true}
        label="Passive Button"
        m="2xs"
        mb="2xl"
      />
      <Text p="2xs" variant="captionMedium">
        Tertiary Variant
      </Text>
      <Button
        size="l"
        variant="tertiary"
        disabled={true}
        label="Passive Button"
        m="2xs"
      />
      <Button
        size="m"
        variant="tertiary"
        disabled={true}
        label="Passive Button"
        m="2xs"
      />
      <Button
        size="s"
        variant="tertiary"
        disabled={true}
        label="Passive Button"
        m="2xs"
      />
    </Box>
  </>
);

export const PressedStates: ButtonStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Pressed States
    </Text>
    <Box mt="xl">
      <Text p="2xs" variant="captionMedium">
        Primary & Secondary Variants
      </Text>
      <Button isPressed={true} label="Save" icon="info" m="2xs" />
      <Button
        isPressed={true}
        kind="neutral"
        label="Save"
        icon="info"
        m="2xs"
      />
      <Button
        isPressed={true}
        kind="success"
        label="Save"
        icon="info"
        m="2xs"
      />
      <Button
        isPressed={true}
        kind="danger"
        label="Save"
        icon="info"
        m="2xs"
        mb="2xl"
      />
      <Text p="2xs" variant="captionMedium">
        Tertiary Variant
      </Text>
      <Button
        variant="tertiary"
        isPressed={true}
        label="Save"
        icon="info"
        m="2xs"
      />
      <Button
        variant="tertiary"
        isPressed={true}
        kind="neutral"
        label="Save"
        icon="info"
        m="2xs"
      />
      <Button
        variant="tertiary"
        isPressed={true}
        kind="success"
        label="Save"
        icon="info"
        m="2xs"
      />
      <Button
        variant="tertiary"
        isPressed={true}
        kind="danger"
        label="Save"
        icon="info"
        m="2xs"
      />
    </Box>
  </>
);
