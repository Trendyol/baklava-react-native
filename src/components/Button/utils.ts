import { ColorProps, VariantProps } from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

type ColorVariantProps = ColorProps<Theme>['color'];

export const variantColorSelector = ({
  kind = 'default',
  variant = 'primary',
  isPressed = false,
  disabled = false,
}: {
  kind?: VariantProps<Theme, 'buttonKindVariants'>['variant'];
  variant?: VariantProps<Theme, 'buttonVariants'>['variant'];
  isPressed?: boolean;
  disabled?: boolean;
}) => {
  if (disabled) {
    return {
      color: 'neutralDarker',
      backgroundColor: 'neutralLightest',
      borderColor: 'neutralLightest',
    } as {
      color: ColorVariantProps;
      backgroundColor: ColorVariantProps;
      borderColor: ColorVariantProps;
    };
  }

  const kindColor = {
    default: 'primaryKey',
    neutral: 'neutralDarker',
    success: 'successKey',
    danger: 'dangerKey',
  }[kind];

  const kindHighlightColor = {
    default: 'primaryHighlight',
    neutral: 'neutralDarkest',
    success: 'successHighlight',
    danger: 'dangerHighlight',
  }[kind];

  let variants = isPressed
    ? {
        primary: {
          color: 'white',
          backgroundColor: kindHighlightColor,
          borderColor: kindHighlightColor,
        },
        secondary: {
          color: 'white',
          backgroundColor: kindHighlightColor,
          borderColor: kindHighlightColor,
        },
        tertiary: {
          color: kindHighlightColor,
          backgroundColor: 'neutralLightest',
          borderColor: 'neutralLightest',
        },
      }
    : {
        primary: {
          color: 'white',
          backgroundColor: kindColor,
          borderColor: kindColor,
        },
        secondary: {
          color: kindColor,
          backgroundColor: 'transparent',
          borderColor: kindColor,
        },
        tertiary: {
          color: kindColor,
          backgroundColor: 'transparent',
          borderColor: 'transparent',
        },
      };

  return variants[variant] as {
    color: ColorVariantProps;
    backgroundColor: ColorVariantProps;
    borderColor: ColorVariantProps;
  };
};
