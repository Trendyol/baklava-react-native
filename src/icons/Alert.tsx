import * as React from 'react';
import Svg, { SvgProps, Path } from 'react-native-svg';

const SvgAlert = (props: SvgProps) => (
  <Svg
    width={24}
    height={24}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    className=""
    {...props}>
    <Path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm-1.53 7.067A1 1 0 0 1 11.468 6h1.065a1 1 0 0 1 .997 1.067l-.4 6a1 1 0 0 1-.998.933h-.264a1 1 0 0 1-.998-.933l-.4-6ZM12 18a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
      fill={props?.fill ?? `#273142`}
    />
  </Svg>
);

export default SvgAlert;
