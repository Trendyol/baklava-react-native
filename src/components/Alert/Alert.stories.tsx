import { ComponentMeta, ComponentStory } from '@storybook/react-native';
import React, { useState } from 'react';
import Text from '../Text/Text';
import Alert from './Alert';
import Button from '../Button/Button';
import Box from '../Box/Box';
import { VariantProps } from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

const variantList = ['info', 'warning', 'success', 'danger'];

const AlertMeta: ComponentMeta<typeof Alert> = {
  title: 'Alert',
  component: Alert,
  argTypes: {
    variant: {
      options: variantList,
      control: { type: 'radios' },
    },
  },
  args: {
    variant: variantList[0] as VariantProps<Theme, 'alertVariants'>['variant'],
    caption: 'Informational Caption',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed tincidunt tellus, tellus non facilisis. Est at ante amet eget ut blandit.',
    closable: true,
    icon: true,
    closed: false,
  },
};

export default AlertMeta;

type AlertStory = ComponentStory<typeof Alert>;

export const Basic: AlertStory = args => {
  const [state, setState] = useState(args);

  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        Alert
      </Text>
      <Box px="m" py="2xs">
        <Alert {...state} onClose={() => setState({ ...state, closed: true })}>
          <Button label="Button" />
          <Button label="Button" variant="secondary" ml="2xs" />
        </Alert>
      </Box>
    </>
  );
};

export const Variants: AlertStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Alert Variants
    </Text>
    <Box px="m" py="2xs">
      <Alert
        caption="Informational Caption"
        description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed tincidunt tellus, tellus non facilisis. Est at ante amet eget ut blandit."
        closable={false}>
        <Button label="Button" />
        <Button label="Button" variant="secondary" ml="2xs" />
      </Alert>
    </Box>
    <Box px="m" py="2xs">
      <Alert
        caption="Informational Caption"
        description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed tincidunt tellus, tellus non facilisis. Est at ante amet eget ut blandit."
        variant="warning"
        closable={true}
        icon={false}>
        <Button label="Button" />
      </Alert>
    </Box>
    <Box px="m" py="2xs">
      <Alert
        caption="Informational Caption"
        description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed tincidunt tellus, tellus non facilisis. Est at ante amet eget ut blandit."
        variant="success"
        closable={false}
        icon={false}>
        <Button label="Button" variant="secondary" />
      </Alert>
    </Box>
    <Box px="m" py="2xs">
      <Alert
        description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed tincidunt tellus, tellus non facilisis. Est at ante amet eget ut blandit."
        variant="danger"
        closable={true}
      />
    </Box>
  </>
);
