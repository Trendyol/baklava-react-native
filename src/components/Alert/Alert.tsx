import React from 'react';
import { Pressable, StyleSheet } from 'react-native';
import {
  VariantProps,
  createRestyleComponent,
  createVariant,
} from '@ergenekonyigit/restyle';
import Box from '../Box/Box';
import Text from '../Text/Text';
import Icon from '../Icon/Icon';
import { IconNameType } from '../Icon/types';
import theme, { Theme } from '../../theme';
import { AlertIcon } from './AlertIcon';
import { deviceWidth } from '../../utils/dimentions';

type AlertProps = React.ComponentProps<typeof Box> & {
  variant?: VariantProps<Theme, 'alertVariants'>['variant'];
  icon?: IconNameType | boolean | null;
  caption?: string | null;
  description?: string;
  closable?: boolean;
  closed?: boolean;
  onClose?: () => void;
  children?: React.ReactNode;
};

const variantVariant = createVariant<Theme, 'alertVariants', 'variant'>({
  property: 'variant',
  themeKey: 'alertVariants',
});

const AlertContainer = createRestyleComponent<
  AlertProps & VariantProps<Theme, 'alertVariants'>,
  Theme
>([variantVariant], Box);

const Alert = ({
  variant = 'info',
  icon,
  caption,
  description,
  closable,
  closed,
  onClose,
  children,
  ...rest
}: AlertProps) => {
  if (closed) {
    return null;
  }

  return (
    <AlertContainer variant={variant} {...rest}>
      <Box p="m" flexDirection="row" width={deviceWidth - theme.spacing['2xl']}>
        <Box pr="2xs">
          <AlertIcon variant={variant} icon={icon} />
        </Box>
        <Box flex={1}>
          {caption ? (
            <Text
              variant="subtitle03Medium"
              pb="2xs"
              testID="alert-caption"
              textAlign={'left'}>
              {caption}
            </Text>
          ) : null}

          {description ? (
            <Text testID="alert-description" textAlign={'left'}>
              {description}
            </Text>
          ) : null}

          {children ? (
            <Box flexDirection="row" pt="2xs">
              {children}
            </Box>
          ) : null}
        </Box>
        <Box>
          {closable ? (
            <Pressable
              onPress={onClose}
              style={styles.closeButton}
              testID="alert-close-button">
              <Icon size="xs" name="close" />
            </Pressable>
          ) : null}
        </Box>
      </Box>
    </AlertContainer>
  );
};

const styles = StyleSheet.create({
  closeButton: {
    padding: theme.spacing['2xs'],
    marginTop: theme.spacing['-2xs'],
    marginRight: theme.spacing['-2xs'],
  },
});

export default Alert;
