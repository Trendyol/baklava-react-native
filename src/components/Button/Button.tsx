import React from 'react';
import { ColorValue, Pressable } from 'react-native';
import {
  backgroundColor,
  border,
  createRestyleComponent,
  createVariant,
  layout,
  LayoutProps,
  spacing,
  spacingShorthand,
  SpacingProps,
  SpacingShorthandProps,
  VariantProps,
} from '@ergenekonyigit/restyle';
import Text, { TextVariants } from '../Text/Text';
import { useIsPressed } from './hooks';
import { IconNameType } from '../Icon/types';
import Icon from '../Icon/Icon';
import { Theme } from '../../theme';
import Box from '../Box/Box';
import { variantColorSelector } from './utils';

export type ButtonProps = React.ComponentProps<typeof Box> &
  React.ComponentProps<typeof Pressable> & {
    children?: React.ReactNode;
    onPress?: () => void;
    label?: string;
    tintColor?: ColorValue;
    icon?: IconNameType | null;
    disabled?: boolean;
    isPressed?: boolean;
    filled?: boolean;
    size?: VariantProps<Theme, 'buttonSizeVariants'>['variant'];
    kind?: VariantProps<Theme, 'buttonKindVariants'>['variant'];
    variant?: VariantProps<Theme, 'buttonVariants'>['variant'];
  };

const sizeVariant = createVariant<Theme, 'buttonSizeVariants', 'size'>({
  property: 'size',
  themeKey: 'buttonSizeVariants',
});

const kindVariant = createVariant<Theme, 'buttonKindVariants', 'kind'>({
  property: 'kind',
  themeKey: 'buttonKindVariants',
});

const variantVariant = createVariant<Theme, 'buttonVariants', 'variant'>({
  property: 'variant',
  themeKey: 'buttonVariants',
});

const ButtonContainer = createRestyleComponent<
  ButtonProps &
    VariantProps<Theme, 'buttonSizeVariants'> &
    VariantProps<Theme, 'buttonKindVariants'> &
    VariantProps<Theme, 'buttonVariants'>,
  Theme
>(
  [
    layout,
    spacing,
    border,
    backgroundColor,
    sizeVariant,
    kindVariant,
    variantVariant,
  ],
  Box,
);

type PressableProps = React.ComponentProps<typeof Pressable> &
  LayoutProps<Theme> &
  SpacingProps<Theme> &
  SpacingShorthandProps<Theme>;

const PressableContainer = createRestyleComponent<PressableProps, Theme>(
  [layout, spacing, spacingShorthand],
  Pressable,
);

const Button = ({
  size = 'm',
  kind = 'default',
  variant = 'primary',
  icon,
  label,
  tintColor,
  disabled = false,
  filled = false,
  ...rest
}: ButtonProps) => {
  const { pressableProps, isPressed } = useIsPressed(rest.isPressed);

  /* istanbul ignore next */
  const onPressIn = () => pressableProps.onPressIn();
  /* istanbul ignore next */
  const onPressOut = () => pressableProps.onPressOut();

  const buttonHorizontalPadding: SpacingProps<Theme>['paddingHorizontal'] =
    label
      ? (
          {
            s: 'm',
            m: 'xl',
            l: '2xl',
          } as Readonly<
            Record<
              keyof Omit<Theme['buttonSizeVariants'], 'defaults'>,
              keyof Theme['spacing']
            >
          >
        )[size]
      : (
          {
            s: '2xs',
            m: 'xs',
            l: 's',
          } as Readonly<
            Record<
              keyof Omit<Theme['buttonSizeVariants'], 'defaults'>,
              keyof Theme['spacing']
            >
          >
        )[size];

  const textVariant: TextVariants = (
    {
      s: 'subtitle03Medium',
      m: 'subtitle03Medium',
      l: 'subtitle02Medium',
    } as const
  )[size];

  const variantColors = variantColorSelector({
    variant,
    kind,
    isPressed,
    disabled,
  });

  const typeColor = disabled
    ? 'neutralLighter'
    : isPressed && variant !== 'tertiary'
    ? 'neutralFull'
    : variantColors.color;

  const iconView = React.useMemo(() => {
    if (!icon) {
      return null;
    }
    if (variant === 'transparent') {
      return (
        <Icon name={icon} size="s" color={tintColor} testID="button-icon" />
      );
    }
    return <Icon name={icon} size="s" color={typeColor} testID="button-icon" />;
  }, [icon, typeColor, tintColor, variant]);

  const labelView = React.useMemo(() => {
    if (!label) {
      return null;
    }
    return (
      <Text
        variant={textVariant}
        color={typeColor}
        marginLeft={icon ? '2xs' : 'none'}
        testID="button-text">
        {label}
      </Text>
    );
  }, [icon, label, textVariant, typeColor]);

  return (
    <PressableContainer
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      disabled={disabled}
      {...rest}>
      <ButtonContainer
        size={size}
        kind={kind}
        // @ts-ignore
        variant={variant}
        paddingHorizontal={buttonHorizontalPadding}
        flexDirection="row"
        alignItems="center"
        justifyContent="center"
        alignSelf={filled ? 'stretch' : 'flex-start'}
        {...variantColors}>
        {iconView}
        {labelView}
      </ButtonContainer>
    </PressableContainer>
  );
};

export default Button;
