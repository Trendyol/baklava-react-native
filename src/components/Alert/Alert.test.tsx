import React from 'react';
import theme from '../../theme';
import { fireEvent, render } from '../../test-utils';
import Alert from './Alert';
import Button from '../Button/Button';

describe('Alert', () => {
  test('should render alert correctly', () => {
    // when
    const { toJSON } = render(
      <Alert caption="caption" description="description" closable={true} />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render null when closed true', () => {
    // when
    const { toJSON } = render(
      <Alert
        caption="caption"
        description="description"
        closable={true}
        closed={true}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render alert with given variant', () => {
    // when
    const { getByTestId } = render(<Alert testID="alert" variant="success" />);

    const alertComponent = getByTestId('alert');

    // then
    expect(alertComponent.props.style[1][0].backgroundColor).toBe(
      theme.colors.successContrast,
    );
  });

  test('should render alert with given children ', () => {
    // given
    const mockOnPress = jest.fn();

    // when
    const { getByTestId } = render(
      <Alert testID="alert" description="description">
        <Button
          label="Button"
          testID="alert-child-button"
          onPress={mockOnPress}
        />
      </Alert>,
    );
    const alertChildButton = getByTestId('alert-child-button');

    fireEvent.press(alertChildButton);

    // then
    expect(mockOnPress).toBeCalled();
  });
});
