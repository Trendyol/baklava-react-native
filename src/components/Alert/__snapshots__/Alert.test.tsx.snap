// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render alert correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#FEF2E8",
          "borderColor": "#F27A1A",
          "borderRadius": 8,
          "borderWidth": 1,
        },
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "padding": 16,
          "width": 718,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingRight": 8,
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="#F27A1A"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        testID="alert-icon"
        title="info"
        vbHeight={24}
        vbWidth={24}
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={
            Object {
              "type": 0,
              "value": 4294081050,
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm-1.5 11.5a1.5 1.5 0 0 1 3 0v5a1.5 1.5 0 0 1-3 0v-5ZM12 6a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3Z"
            fill={
              Object {
                "type": 0,
                "value": 4294081050,
              }
            }
            fillRule={0}
            propList={
              Array [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessible={true}
        style={
          Array [
            Object {
              "color": "#273142",
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 16,
              "paddingBottom": 8,
            },
          ]
        }
        testID="alert-caption"
      >
        caption
      </Text>
      <Text
        accessible={true}
        style={
          Array [
            Object {
              "color": "#273142",
              "fontFamily": "Rubik-Regular",
              "fontSize": 14,
            },
          ]
        }
        testID="alert-description"
      >
        description
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": -8,
            "marginTop": -8,
            "padding": 8,
          }
        }
        testID="alert-close-button"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="14"
          bbWidth="14"
          fill="#273142"
          focusable={false}
          height={14}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 14,
                "width": 14,
              },
            ]
          }
          testID="icon"
          title="close"
          vbHeight={24}
          vbWidth={24}
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={
              Object {
                "type": 0,
                "value": 4280758594,
              }
            }
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              clipRule={0}
              d="M.54.54a1.846 1.846 0 0 1 2.612 0L12 9.39 20.848.54a1.846 1.846 0 0 1 2.611 2.612L14.611 12l8.848 8.848a1.846 1.846 0 0 1-2.61 2.611L12 14.611l-8.848 8.848A1.846 1.846 0 0 1 .54 20.85L9.389 12 .541 3.152A1.846 1.846 0 0 1 .54.54Z"
              fill={
                Object {
                  "type": 0,
                  "value": 4280758594,
                }
              }
              fillRule={0}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Alert should render null when closed true 1`] = `null`;
